java/src/jmri/jmrix/bachrus/SpeedoConsoleFrame.java

```diff
@@ -306,22 +433,29 @@ public void initComponents() {
         scalePanel.setLayout(new FlowLayout());
 
         scaleList.setToolTipText(Bundle.getMessage("SelectScaleToolTip"));
-        try {
-            scaleList.setSelectedItem(prefs.getComboBoxLastSelection(selectedScalePref));
-        } catch (ArrayIndexOutOfBoundsException e) {
-            scaleList.setSelectedItem(defaultScale);
+        String lastSelectedScale = prefs.getComboBoxLastSelection(selectedScalePref);
+        if (lastSelectedScale != null && !lastSelectedScale.equals("")) {
+            try {
+                scaleList.setSelectedItem(lastSelectedScale);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                scaleList.setSelectedIndex(defaultScale);
+            }
+        } else {
+            scaleList.setSelectedIndex(defaultScale);
         }
-        if ( scaleList.getSelectedIndex() > -1 ) {
+
+        if (scaleList.getSelectedIndex() > -1) {
             selectedScale = scales[scaleList.getSelectedIndex()];
         }
+
         // Listen to selection of scale
-        scaleList.addActionListener(new java.awt.event.ActionListener() {
-            @SuppressWarnings("unchecked") // action semantics pass an Object that must be a JComboBox<String>
-            @Override
-            public void actionPerformed(java.awt.event.ActionEvent e) {
+        scaleList.addActionListener(e -> {
+            try {
                 JComboBox<String> cb = (JComboBox<String>) e.getSource();
                 selectedScale = scales[cb.getSelectedIndex()];
```

Comment:
Since `scaleList` is the source of the event here, you can rewrite this listener as:

```
selectedScale = scales[scaleList.getSelectedIndex()];
checkCustomScale();
```

and remove the try/catch statement

Reply:
Thanks for the simple solution! I've made the suggested change and committed it to the pull request's branch.